src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -185,7 +185,7 @@ public IPAddress(ReadOnlySpan<byte> address)
         {
             if (address.Length == IPAddressParserStatics.IPv4AddressBytes)
             {
-                PrivateAddress = (uint)((address[3] << 24 | address[2] << 16 | address[1] << 8 | address[0]) & 0x0FFFFFFFF);
```

Comment:
What a strange `0x0FFFFFFFF` masking.  I have to wonder what the original author was thinking there.